src/worker/mod.rs

```diff
@@ -6,7 +6,7 @@ pub use crate::worker::config::{WorkerConfig, WorkerConfigBuilder};
 pub(crate) use dispatcher::WorkerDispatcher;
 
 use crate::{
-    errors::{CompleteWfError, WorkflowUpdateError},
+    errors::CompleteWfError,
```

Comment:
Looks like this is the only error that can be returned by worker methods.
After skimming the code I'm not convinced we evict in that case, can you verify?
One instance I think that might be missed is here:
https://github.com/temporalio/sdk-core/blob/1e92f51a19d80ba0b3124906d810471e3bad61bd/src/worker/mod.rs#L652

And possibly here:
https://github.com/temporalio/sdk-core/blob/1e92f51a19d80ba0b3124906d810471e3bad61bd/src/worker/mod.rs#L324

This is the TS side of completion:
https://github.com/temporalio/sdk-typescript/blob/53cffb41a1edc7f23a1676c5d1e8cee2421bf4ba/packages/worker/src/worker.ts#L806

Reply:
Yup, eviction happens there too (and is the most covered path by tests actually)